shell.nix

```
@@ -9,20 +9,40 @@ let
   pkgsGhc = import sources.nixpkgs-ghc8107 {};
   build_daml = import ./nix/daml.nix;
   packell = import ./nix/packell.nix;
+
   damlYaml = builtins.fromJSON (builtins.readFile (pkgs.runCommand "daml.yaml.json" { yamlFile = ./daml.yaml; } ''
-                ${pkgs.yj}/bin/yj < "$yamlFile" > $out
-              ''));
-  daml = (build_daml { stdenv = pkgs.stdenv;
-                       jdk = pkgs.openjdk11_headless;
-                       sdkVersion = damlYaml.sdk-version;
-                       damlVersion = damlYaml.daml-version;
-                       tarPath = damlYaml.daml-tar-path or null;
-                       curl = pkgs.curl;
-                       curl_cert = "${pkgs.cacert}/etc/ssl/certs/ca-bundle.crt";
-                       os = if pkgs.stdenv.isDarwin then "macos" else "linux";
-                       osJFrog = if pkgs.stdenv.isDarwin then "macos" else "linux-intel";
-                       hashes = { linux = "JYJ6pOsJf+m3ymForJO54dVTgLS0lyXoxOA6YOck0KY=";
-                                  macos = "fH3ZS5h+O2w2F3oeelXBAmc2BHfhOzIaefjVvUjryWk="; };});
+    ${pkgs.yj}/bin/yj < "$yamlFile" > $out
+  ''));
+
+  os =
+    if pkgs.stdenv.isDarwin then "macos" else
+    if pkgs.stdenv.isLinux then "linux" else
+    throw "Unsupported OS";
+
+  arch =
+    if pkgs.stdenv.isDarwin then "x86_64" else
```

Adding context to why we are setting `x86_64` for future reference:

```suggestion
    if pkgs.stdenv.isDarwin then "x86_64" else # Daml SDK on macOS is only available on x86 architecture
```